### DISCUSSION TITLE
HOW do I configure a disk to use???

### DISCUSSION THREAD

**Discussion by jongravois** [spatie/laravel-medialibrary#2367] [Q&A]
[Associating Files](https://spatie.be/docs/laravel-medialibrary/v9/basic-usage/associating-files)

On this page of the docs, I read

```
You can associate a file with a model like this:

$yourModel = YourModel::find(1);
$yourModel
   ->addMedia($pathToFile)
   ->toMediaCollection();
The file will now be associated with the YourModel instance and will be moved to the **disk you've configured**.
```

I have searched the docs and I can't find **how you configure a disk to use**.

Thoughts???

**Comment by martin-ro** [comment]
You can publish the config with `php artisan vendor:publish --provider="Spatie\MediaLibrary\MediaLibraryServiceProvider" --tag="config"` and set in your `.env` the disk. For example setting the disk to S3 use: `MEDIA_DISK=s3`.

You can also set the disk on your model to store only specific collections to specific disks:
```php
/**
 * Register media collections.
 */
public function registerMediaCollections(): void
{
    $this->addMediaCollection('photos')->useDisk('s3');
    $this->addMediaCollection('avatars')->useDisk('public');
}
```
